src/Hamiltonians/Molecular.jl

```diff
@@ -0,0 +1,780 @@
+        return MolecularHamiltonianOffDiagonalsIteratorState((0, 2), from_tuple, to_tuple)
+    elseif (
+        iter.excitation_prefix_sums[3] + 1 <= iter_index <= iter.excitation_prefix_sums[4]
+    ) # 2 electron excitation in alpha channel
+        iter_offset = iter_index - iter.excitation_prefix_sums[3] - 1
+        from = iter_offset ÷ binomial(length(iter.modes.unoccupied[1]), 2) + 1
+        to = iter_offset % binomial(length(iter.modes.unoccupied[1]), 2) + 1
+        from_tuple = unrank_combination(length(iter.modes.occupied[1]), from)
+        to_tuple = unrank_combination(length(iter.modes.unoccupied[1]), to)
+        return MolecularHamiltonianOffDiagonalsIteratorState((2, 0), from_tuple, to_tuple)
+    elseif (
+        iter.excitation_prefix_sums[4] + 1 <= iter_index <= iter.excitation_prefix_sums[5]
+    ) # 1 electron excitation in both channels
+        iter_offset = iter_index - iter.excitation_prefix_sums[4] - 1
+        alpha_from =
+            iter_offset ÷ (
+                length(iter.modes.unoccupied[1]) *
+                length(iter.modes.occupied[2]) *
+                length(iter.modes.unoccupied[2])
+            )
+        iter_offset -=
+            alpha_from * (
+                length(iter.modes.unoccupied[1]) *
+                length(iter.modes.occupied[2]) *
+                length(iter.modes.unoccupied[2])
+            )
+        alpha_to =
+            iter_offset ÷
+            (length(iter.modes.occupied[2]) * length(iter.modes.unoccupied[2]))
+        iter_offset -=
+            alpha_to * (length(iter.modes.occupied[2]) * length(iter.modes.unoccupied[2]))
+        beta_from = iter_offset ÷ length(iter.modes.unoccupied[2])
+        beta_to = iter_offset % length(iter.modes.unoccupied[2])
+        return MolecularHamiltonianOffDiagonalsIteratorState(
+            (1, 1), (alpha_from + 1, beta_from + 1), (alpha_to + 1, beta_to + 1)
+        )
+    end
+    return MolecularHamiltonianOffDiagonalsIteratorState((1, 1), (0, 0), (0, 0))
+end
+
+"""
+    unrank_combination(n::Int, i::Int)
+
+Return the 2-element combination (in lex order) of set {1,...,n} with index i.
+When generating the excitation state with 2 electrons excited in a single spin
+channel, we use the indice of array storing (un)occupied mode maps. For example,
+we select 2 non-repeating indices of array storing unoccupied mode maps to
+represent modes the electrons should go to as `to_unoccupieds` tuple in
+[`MolecularHamiltonianOffDiagonalsIteratorState`](@ref).
+If there are 4 unoccupied modes, all the possible combination of chosen in
+lexicographic order should be:
+
+    index:       1      2      3      4      5      6
+    combination: (1, 2) (1, 3) (1, 4) (2, 3) (2, 4) (3, 4)
+
+The length of all the combinations is ``4C2 = 6``.
+Given the index(`i`), to determin the value of 1st position(`x`) in the
+combination, we can first determine how many combinations when `x` is
+from 1 to `n`, the number is calculated by ``(n-x)C1 = (n-x)``.
```

Comment:
```suggestion
The length of all the combinations is ``\\binom{4}{2} = 6``.
Given the index(`i`), to determine the value of the first position(`x`) in the
combination, we can first determine the number of combinations when `x` goes
from 1 to `n`. The number is calculated by ``\\binom{n-x}{1} = n-x``.
```
You'll need to check whether the `amsmath` package of latex is supported. Otherwise you can leave it as you had it.